exceptionHandling/Task2/NumberParser.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+namespace Task2
+{
+    public class NumberParser : INumberParser
+    {
+        public int Parse(string stringValue)
+        {
+            if (stringValue is null)
+            {
+                throw new ArgumentNullException(nameof(stringValue),"Value is null");
+
+            }
+            var str = stringValue.Trim();
+
+            if (string.IsNullOrWhiteSpace(str) || string.IsNullOrEmpty(str))
+            {
+                throw new FormatException("Invalid input format");
+            }
+            
+            var num = 0;
+            var sign = 1;
+            int i = 0;
+            int res = 0;
+            if (str[0] == '-')
+            {
+                sign = -1;
+                i = 1;
+            }
+            if (str[0] == '+')
+            {
+                sign = 1;
+                i = 1;
+            }
+            
+            while (i < str.Length)
+            {
+                int currentNum = str[i] - '0';
+                if (currentNum > 9 || currentNum < 0)
+                {
+                    throw new FormatException("Input must contain only digits");
+                }
+
+                num = (num * 10) + currentNum;
+                i++;
+            }
+
+            res = num * sign;
+            
+            if (res > Int32.MaxValue || res < int.MinValue)
```

Comment:
The reason why tests are failing is in 'num' variable. By default C# compiler won't throw an exception if you try to overflow/underflow the variable, it will simply 'reset the counter', if I can call it so ( e. g. int.MaxValue + 1 = int.MinValue), so it will never become overflowed/underflowed and since you declared 'num' as _int_ this statement will always be _false_. 

To resolve this you can use built-in _checked_ keyword in the place you expect overflow (line: 44 most likely), it will tell the compiler to throw the exception if overflow happens. Such way you will not need this if statement at all. More details here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/checked-and-unchecked.
Or, you can declare num as _long_, so it can store twice larger values as int and this if will work as expected.